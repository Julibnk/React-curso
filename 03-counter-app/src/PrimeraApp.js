// FC (Functional component)

import PropTypes from 'prop-types';

const PrimeraApp = ({ saludo, subtitulo }) => {


    return (
        <>
            <h1>{ saludo }</h1>
            <p>{ subtitulo }</p>
        </>
    );
       
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    subtitulo: 'Subtitulo'
}

export default PrimeraApp;