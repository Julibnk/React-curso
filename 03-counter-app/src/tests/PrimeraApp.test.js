import PrimeraApp from "../PrimeraApp";
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

describe('Pruebas en <PrimeraApp />', () => {
    // test('debe mostrar el mensaje Hola sou Julian ', () => {

    // Test sin enzyme
    //     const saludo = 'Hola soy Julian';

    //     const { getByText } = render(<PrimeraApp saludo={saludo}/>);

    //     expect( getByText( saludo ) ).toBeInTheDocument()
    // })

    test('debe mostrar <PrimeraApp /> correctamente', () => {

        const saludo = 'Hola soy Jul';
        const wrapper = shallow( <PrimeraApp saludo={saludo} />);

        expect( wrapper ).toMatchSnapshot();

    })
    
    test('debe mostrar el subtitulo enviado por props', () => {

        const saludo = 'Hola soy Jul';
        const subtitulo = 'Soy un subtitulo';
        const wrapper = shallow( <PrimeraApp saludo={saludo} subtitulo={subtitulo} />);

        const textoParrafo = wrapper.find('p').text();

        expect(textoParrafo).toBe(subtitulo);

    })

}) 
