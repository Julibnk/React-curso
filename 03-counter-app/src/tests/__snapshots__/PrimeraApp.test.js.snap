// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> debe mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola soy Jul
  </h1>
  <p>
    Subtitulo
  </p>
</Fragment>
`;
